src/pconn.cc

```diff
@@ -584,14 +583,14 @@ PconnModule::remove(PconnPool *aPool)
 void
 PconnModule::dump(StoreEntry *e)
 {
-    typedef Pools::const_iterator PCI;
-    int i = 0; // TODO: Why number pools if they all have names?
-    for (PCI p = pools.begin(); p != pools.end(); ++p, ++i) {
+    PackableStream yaml(*e);
+    IfNoOutput defaultText("{}\n", yaml); // print if no pools are defined
+
+    for (const auto &p: pools) {
         // TODO: Let each pool dump itself the way it wants to.
-        storeAppendPrintf(e, "\n Pool %d Stats\n", i);
-        (*p)->dumpHist(e);
-        storeAppendPrintf(e, "\n Pool %d Hash Table\n",i);
-        (*p)->dumpHash(e);
+        yaml << defaultText  << "pool " << p->description() << ":\n";
```

Comment:
Please report PconnPool::count() for every pool, including empty pools.

```yaml
  open connections: 0
```

After this and another change request, please remove defaultText and (highly questionable) IfNoOutput from this PR.

Reply:
Sounds good, it's a better solution to the problem I was trying to solve